FreeRTOS/FreeRTOS
# DNS: MISRA + Coverity

This PR focuses on resolving the MISRA compliance related warnings in FreeRTOS_DNS.c. As a side effect, a boolean True/False was defined in the FreeRTOS_IP.h. This header file is chosen since it is included in almost all of the source files and the boolean Macros will be available to all.

This doesn't ensure that it doesn't have any warnings at all since a lot of warnigns require code restructure/rework and some warnings require editing header files and some are portable layer warnings.

Test Steps
-----------
Ran all Microchip Curiosity Tests on CI. Thereby ensuring buildability and executability of the code. GGD tests failed.
Also ran CBMC proofs.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
